app/src/main/java/ar/com/wolox/android/login/ui/training/TrainingFragment.java

```diff
@@ -0,0 +1,68 @@
+package ar.com.wolox.android.login.ui.training;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import ar.com.wolox.android.R;
+import ar.com.wolox.android.login.ui.viewpager.ViewpagerActivity;
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+
+public class TrainingFragment extends WolmoFragment<LoginPresenter> implements ILoginView {
+
+    @BindView(R.id.fragment_login_email) TextView mEmail;
+    @BindView(R.id.fragment_login_password) TextView mPassword;
+
+    @Inject
+    public TrainingFragment() {}
+
+    @Override
+    public int layout() {
+        return R.layout.fragment_login;
+    }
+
+    @Override
+    public void init() {
+        mEmail.setText(getPresenter().getEmail());
+        mPassword.setText(getPresenter().getPassword());
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+    }
+
+    @OnClick(R.id.fragment_login_login)
+    public void onLogin() {
+        getPresenter().login(mEmail.getText().toString(), mPassword.getText().toString());
+    }
+
+    @Override
+    public void onLoginSuccess() {
+
+    }
+
+    @Override
+    public void onLoginEmailError(String error) {
```

Comment:
Here instead of receiving the error message you can receive an `enum` or an error class and get the error message from `strings.xml`.

Reply:
It's done in the next branch without taking the message from strings.xml so I will fix that in the next PR. (https://github.com/wolox-training/dm-android/tree/login-frontend-2/app/src/main/java/ar/com/wolox/android/training/ui/errors)